src/dlr_relayvm.cc

```diff
@@ -0,0 +1,258 @@
+#include "dlr_relayvm.h"
+
+
+#include <stdlib.h>
+#include <fstream>
+#include <iterator>
+#include <numeric>
+
+using namespace dlr;
+
+void RelayVMModel::InitModelPath(std::vector<std::string> paths) {
+  path_ = std::make_unique<ModelPath>();
+  std::vector<std::string> paths_vec;
+  for (auto path : paths) {
+    ListDir(path, paths_vec);
+  }
+
+  for(auto path: paths_vec) {
+    if (!EndsWith(path, LIBDLR) && EndsWith(path, ".so")) {
+      path_->model_lib = path;
+    } else if (EndsWith(path, ".ro")) {
+      path_->relay_executable = path;
+    } else if (EndsWith(path, ".meta")) {
+      path_->metadata = path;
+    }
+  }
+
+  if (path_->model_lib.empty() || path_->relay_executable.empty() || path_->metadata.empty()) {
+    LOG(FATAL) << "Invalid model artifact!";
```

Comment:
Metadata should be allowed to be empty.

Reply:
I have the same concern here, I noticed in dlr_tvm.cc has the same restriction that the metadata exists and not empty. Sorry if I miss any discussion but could you give me some background or link for previous discussion.

Reply:
I didn't see any apis to get input/output types and name in VM hence to keep the APIs consistent with the other implementation we are relying on the presence of metadata file.

Reply:
We can't get input types(shape, dtype) until we set the actual input due to the dynamism.

Reply:
:+1: We can only get the number of channels from metadata file.